src/Compilers/Core/Portable/EncodedStringText.cs

```diff
@@ -74,7 +86,7 @@ internal static SourceText Create(Stream stream, Encoding defaultEncoding = null
 
             try
             {
-                return Decode(stream, defaultEncoding ?? s_defaultEncoding, checksumAlgorithm, throwIfBinaryDetected: detectEncoding);
+                return Decode(stream, defaultEncoding ?? GetDefaultAnsiEncoding(), checksumAlgorithm, throwIfBinaryDetected: detectEncoding);
```

Comment:
Our goal for Determinism is to make the compiler deterministic, not merely to make it possible to use the compiler in a deterministic way. In other words the compiler should be deterministic whether or not you use particular command-line options.

I think we need to consider any environmental inputs (e.g. environment variables and default code page) as part of the input to the compiler for the purpose of determinism.